db/log_reader.cc

```diff
@@ -450,17 +461,49 @@ unsigned int Reader::ReadPhysicalRecord(Slice* result, size_t* drop_size) {
 
     buffer_.remove_prefix(header_size + length);
 
-    *result = Slice(header + header_size, length);
-    return type;
+    if (uncompress_ && type != kSetCompressionType) {
+      // Uncompress compressed records
+      uncompressed_record_.clear();
+      size_t uncompressed_size = 0;
+      int remaining = 0;
+      do {
+        remaining =
+            uncompress_->Uncompress(header + header_size, length,
+                                    uncompressed_buffer_, &uncompressed_size);
+        if (uncompressed_size > 0) {
+          uncompressed_record_.append(uncompressed_buffer_, uncompressed_size);
+        }
+      } while (remaining > 0 || uncompressed_size == kBlockSize);
```

Comment:
Why do we need to check both? Isn't ```remaining > 0``` sufficient?

Reply:
I think I had to do that because in some case remaining was 0 but there was still uncompressed chunks left. Let me remove that and see if it makes a difference.

Reply:
Fixed.